tests/test_table.py

```diff
@@ -2696,6 +2696,58 @@ def test_subsample_md_copy_bug(self):
         obs = table.subsample(5)
         self.assertEqual(obs, exp)
 
+    def test_subsample_byid_with_replacement(self):
+        dt = Table(np.array([0,1,2], [2,0,1], [1, 2, 0]),
+                   ['O1', 'O2', 'O3'],
+                   ['S1', 'S2', 'S3'])
+        with self.assertRaises(ValueError):
+            dt.subsample(20, by_id=True, with_replacement=True)
+
+    def test_subsample_without_replacement_unique_results(self):
+        """
+        As in scikit-bio. Given a vector of observations, the total number of
+        unique subsamplings when n == sum(vector) - 1 should be equal to the
+        number of unique categories of observations when the vector is
+        subsampled without replacement. If the vector is subsamples *with*
+        replacement, however, there should be more than 10 different
+        possible subsamplings.
+        """
+        a = np.array([2, 1, 2, 1, 8, 6, 3, 3, 5, 5]).T
+        dt = Table(data=a, sample_ids=['S1',],
+                   observation_ids=['OTU{:02d}'.format(i) for i in range(10)])
+        actual = set()
+        for i in range(1000):
+            obs = dt.subsample(35)
+            actual.add(tuple(obs.data('S1')))
+        self.assertTrue(len(actual) == 10)
+
+    def test_subsample_with_replacement_unique_results(self):
+        """
+        As in scikit-bio. Given a vector of observations, the total number of
+        unique subsamplings when n == sum(vector) - 1 should be equal to the
+        number of unique categories of observations when the vector is
+        subsampled without replacement. If the vector is subsamples *with*
+        replacement, however, there should be more than 10 different
+        possible subsamplings.
+        """
+        a = np.array([2, 1, 2, 1, 8, 6, 3, 3, 5, 5]).T
+        dt = Table(data=a, sample_ids=['S1',],
+                   observation_ids=['OTU{:02d}'.format(i) for i in range(10)])
+        actual = set()
+        for i in range(1000):
+            obs = dt.subsample(35, with_replacement=True)
+            actual.add(tuple(obs.data('S1')))
+        self.assertTrue(len(actual) > 10)
```

Comment:
This should be `self.assertGreaterThan(len(actual), 10)`

Reply:
Running `nosetests` for me throws an error here:

`AttributeError: 'SparseTableTests' object has no attribute 'assertGreaterThan'`

I'm looking into where a declaration needs to be made, but if you know the answer off the top of your head, I'd be grateful to know it too.

Reply:
Nevermind, I found it. It's `assertGreater` rather than `assertGreaterThan`.